lib/new_relic/agent.rb

```diff
@@ -203,6 +204,12 @@ def record_metric(metric_name, value) # THREAD_LOCAL_ACCESS
       agent.stats_engine.tl_record_unscoped_metrics(metric_name, value)
     end
 
+    def record_metric_once(metric_name, value = 0.0)
+      return if @metrics_already_recorded.add?(metric_name)
```

Comment:
`Set#add?` will return the set itself if the addition took place and `nil` otherwise. Given that you only want to record the metric once, I think you should use `return unless` here instead of `return if`.

From looking at your new test, I think the reason it passes is because the 2nd call actually causes the metric to be recorded and then there's no 3rd call which would have caused a 2nd recording. I think you should update the test to perform fully 3 separate `record_metric_once` calls and then have it confirm that only 1 call to `record_metric` is made after the `if -> unless` change is made.

Reply:
thanks for catching that! the way it's working currently is anti-helpful lol. 
updated in 026baa594e6b0e17c408f6cb6cda7c56259f5df1